internal/jimm/role/role_test.go

```diff
@@ -0,0 +1,236 @@
+// Copyright 2024 Canonical.
+package role
+
+import (
+	"context"
+	"sort"
+	"strconv"
+	"testing"
+	"time"
+
+	cofga "github.com/canonical/ofga"
+	qt "github.com/frankban/quicktest"
+	"github.com/frankban/quicktest/qtsuite"
+
+	"github.com/canonical/jimm/v3/internal/common/pagination"
+	"github.com/canonical/jimm/v3/internal/db"
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/errors"
+	"github.com/canonical/jimm/v3/internal/openfga"
+	ofganames "github.com/canonical/jimm/v3/internal/openfga/names"
+	jimmtest "github.com/canonical/jimm/v3/internal/testutils/jimmtest"
+)
+
+type roleManagerSuite struct {
+	manager    *roleManager
+	user       *openfga.User
+	db         *db.Database
+	ofgaClient *openfga.OFGAClient
+}
+
+func (s *roleManagerSuite) Init(c *qt.C) {
+	// Setup DB
+	db := &db.Database{
```

Comment:
since we have the RoleStore inteface, shouldn't we be mocking it in out unit tests?

Reply:
Yes! But I haven't for this as to not stray away massively from what we do currently. I want us to reach a consensus on mocking and how small services like this should look.